client/src/ecash/mod.rs

```diff
@@ -46,13 +46,13 @@ impl EcashWallet {
 
         let spending_conditions = SpendingConditions::new_p2pk(
             self.secret.public_key(),
-            Conditions::new(
+            Some(Conditions::new(
                 Some(user.contract.time_limit),
                 Some(public_keys),
```

Comment:
According to the CDK docs this are "Additional Public keys" which will be considered in adition to self.secret.public_key() above. So we can remove buyer pubkey from this vec. https://docs.rs/cdk/0.1.1/cdk/nuts/nut11/struct.Conditions.html#structfield.pubkeys


```suggestion
                Some(vec![seller_pubkey, provider_pubkey]),
```

Reply:
This probably is even security relevant as the buyer could provide 2 signatures if he has 2 public keys in the contract?

Reply:
Good catch! Moreover, it'd be more sound to set the seller's pubkey in the first argument and the buyer's pubkey in the optional vector. The first argument semantically corresponds to the data field in NUT-11, which is supposed to be the token receiver.